### ISSUE TITLE
bytefmt.ByteSize returns a unsupported string for CF.

### ISSUE THREAD

**Issue by simonjohansson** [cloudfoundry/bytefmt#18]
Howdie!

We have a use-case where we use `code.cloudfoundry.org/cli/util/manifest.ReadAndMergeManifests` and `code.cloudfoundry.org/cli/util/manifest.WriteApplicationManifest` to update a manifest in place in Concourse with secrets from Vault and some extra stuff before doing a CF push and noticed an issue that causes the deployment to fail.

Given a manifest like
```
---
applications:
- name: subjects-backend
  memory: 1536M 
```

after a read and a write the manifest will look like

```
---
applications:
- name: subjects-backend
  memory: 1.5G 
```

Which causes deployment to fail

```
Error reading manifest file:
Invalid value for 'memory': 1.5G
Byte quantity must be an integer with a unit of measurement like M, MB, G, or GB
```

```
type Application struct {
         ...
   	Memory          types.NullByteSizeInMb
        ....
}

func (b NullByteSizeInMb) String() string {
	if !b.IsSet {
		return ""
	}

	return bytefmt.ByteSize(b.Value * bytefmt.MEGABYTE)
}
```

```
package main

import (
	"code.cloudfoundry.org/bytefmt"
	"fmt"
)

func main() {
	memory := "1536M"
	fmt.Println(memory)

	parsed, _ := bytefmt.ToMegabytes(memory)
	fmt.Println(parsed)

	andBack := bytefmt.ByteSize(parsed * bytefmt.MEGABYTE)
	fmt.Println(andBack)
}

// 1536M
// 1536
// 1.5G
```

**Comment by emalm** [comment]
Thanks for the report, @simonjohansson. This issue sounds like something we would need to work with the CLI team to resolve. @xenophex and @abbyachau, can we coordinate about this?

Best,
Eric, CF Diego PM

**Comment by XenoPhex** [comment]
Howdy @simonjohansson!

The main use-case of `bytefmt.ByteSize` is to print the "human readable" form of an memory amount. This means it's designed to convert `1024MB` to `1G`, `1024G` to `1TB`, etc. As the documentation for the function says, `bytefmt.ByteSize` will return `The unit that results in the smallest number greater than or equal to 1 is always chosen.`

My recommendation is to manually format a string after you call `ToMegabytes` (as you already know what the unit should be after the function):
```
package main

import (
	"fmt"

	"code.cloudfoundry.org/bytefmt"
)

func main() {
	memory := "1536M"
	fmt.Println(memory)

	parsed, _ := bytefmt.ToMegabytes(memory)
	fmt.Println(parsed)

	andBack := fmt.Sprintf("%dM", parsed)
	fmt.Println(andBack)
}
// 1536M
// 1536
// 1536M
```

#### An aside on using packages from the CF CLI project
Please note that the CF CLI makes no guarantees [of any sorts] for any packages outside of `code.cloudfoundry.org/cli/plugins`. Internal packages will change without warning and many of them are designed to only work inside the context of the CF CLI. Also, the CF CLI team provides **no support** for any issues that come up from using internal packages. So I will strongly recommend you roll your own code, or copy and maintain your own version.

**TLDR** Please don't use internal CF CLI packages. Thanks!

**Comment by emalm** [comment]
Hey, @simonjohansson,

I had a chance to catch up with @XenoPhex some more about this, and we have a better idea of what's going on. Prior to CF CLI v6.34.0, `cf push` would fail to handle these floating-point memory quantities, apparently because of bytefmt's underlying inability to parse them. With the `cf push` rewrite present in CF CLI v6.34.0, the `cf push` command now does handle them correctly on the `-m` flag or in a manifest. Unfortunately, that rewritten command does not apply to CF CLI plugins, which still get the equivalent of the old `push` command internally, so if you're using a plugin to deploy the app with the new manifest we would still expect it to fail this way.

Anyway, since again this is an issue with the CF CLI that the bytefmt library can't solve in isolation, I'm closing this out. Please feel free to open a new issue on the [CLI repository](https://github.com/cloudfoundry/cli/issues/new) if you'd like to continue discussion there.

Best,
Eric

**Comment by simonjohansson** [comment]
Thanks for taking the time @emalm @XenoPhex 

Ive updated from `6.35.0` to `6.36.2` and now its working!

Point taken about using internal packages, but lazy people gonna be lazy.. :)
